dot/state/epoch_test.go

```diff
@@ -226,10 +226,10 @@ func TestEpochState_GetEpochFromTime(t *testing.T) {
 	require.Equal(t, uint64(99), epoch)
 }
 
-type inMemoryNextEpochData struct {
-	epoch          uint64
-	hashes         []common.Hash
-	nextEpochDatas []types.NextEpochData
+type inMemoryBABEData[T any] struct {
+	epoch  uint64
+	hashes []common.Hash
+	Data   []T
```

Comment:
nit Can we have `Data` unexported as `data`?

Also generics are going wild :tiger: !!

Reply:
I am starting to see why go did not have generics. The code without them looks so much better!